apps/src/templates/teacherDashboard/SectionRow.jsx

```diff
@@ -207,6 +215,12 @@ class SectionRow extends Component {
     } = this.props;
     const { editing, deleting } = this.state;
 
+    // When deleting a section, I've occasionally seen us attempt a render with
+    // a null section for some reason. This is here to provide defense against this.
+    if (!section) {
+      return null;
+    }
```

Comment:
I think you could also avoid this by passing the whole `section` prop in from `SectionTable` instead of just the `sectionId`, removing the state-update/render-ordering dependency you identified here:
```jsx
section: state.teacherSections.sections[ownProps.sectionId],
```
Not sure it's been documented anywhere, but it might be an antipattern for rows of a React table to fetch table data themselves - in any case, libraries like [Reactabular](https://reactabular.js.org/#/) and [Reactable](https://github.com/glittershark/reactable) encourage you to pass data in at the top layer and minimize dependencies in rows.

Pretty sure you can safely leave `SectionRow` connected for the other props the store provides though; `validLoginTypes`, `validGrades`, and `validAssignments` sound like page constants.